dashboard/app/models/activity.rb

```diff
@@ -72,4 +72,49 @@ def Activity.recent(limit)
     # yeah, this is a lot like .last -- but I want a dataset not an array
     Activity.order('id desc').limit(limit)
   end
+
+  # Returns a new Activity and queues an SQS message to asynchronously create it in
+  # the database. Note that the id in the return object will not be populated
+  # because it has not been saved yet. The Activity must pass validation or an
+  # exception will be thrown synchronously.
```

Comment:
would it be possible to generate a guid as the id here so that we know what it will be now, and then the write will use that id later?